### PR TITLE
refactor: Remove PureRenderMixin (deprecated), extend from PureComponent #1287

### PR DESCRIPTION
**Pull request by jurmarcus**
- [x] I have read the [Contributor Guide](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md)
- [x] My PR title and commit messages are in [conventional-changelog-standard](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md#how-should-i-write-my-commit-messages-and-pr-titles) format.

Removed all references to the 'react-addons-pure-render-mixin' and extend the following components from React.PureComponent:

- CodeCell 
- Cell 
- CellCreator 
- CellCreatorButtons 
- Display 
- DraggableCell 
- Editor 
- Inputs 
- MarkdownCell 
- Toolbar 
- LatexRenderer 
- Notebook 
- App 

Also refactored the Inputs component to be a stateless function.

### CODE REVIEW

File: src/notebook/components/cell/inputs.js [hunk 1]
```diff
@@ -1,29 +1,19 @@
...
 type Props = {
   executionCount: any,
   running: boolean,
 };
 
-export default class Inputs extends React.Component {
-  props: Props
-  shouldComponentUpdate: (p: Props, s: void) => boolean;
+export default function Inputs(props: Props): ?React.Element<any> {
+  const { executionCount, running } = props;
```
Comment (peggyrayzis) [inline src/notebook/components/cell/inputs.js:10]: Awesome! Stateless functional components are great and we should try to use more of them where possible in nteract.

**Review by peggyrayzis — CHANGES_REQUESTED** [review]
Thank you for taking this on! Really great work here. Before we merge, could you please reformat your commit messages to [conventional changelog standard](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md#how-should-i-write-my-commit-messages-and-pr-titles)? Please let me know if you need help rebasing interactively to make this happen! 😀

**Comment by jurmarcus** [comment]
Hey! Sorry about that D:, I think the formatting of the commit messages should be now in line with the wanted specification.

**Review by peggyrayzis — APPROVED** [review]

**Comment by peggyrayzis** [comment]
Welcome to the nteract party, @jurmarcusallen! 🎉

**Comment by peggyrayzis** [comment]
Closes #1287

**Pull request merged**